### PR TITLE
CPUDiscovery: remove windows support.

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Comment by ndellingwood** [comment]
Test failure, serial clang/7.0 debug build

```
 AMD64
Compiler: clang C++
AMD64
CMAKE_BUILD_TYPE=Debug COVERAGE=yes 

$ clang --version

clang version 7.0.0 (tags/RELEASE_700/final)

...


[ RUN      ] TEST_CATEGORYDeathTest.view_layoutstride_right_to_layoutleft_assignment

view_layoutstride_right_to_layoutleft_assignment: srand(1578358721)

/home/travis/build/kokkos/kokkos/core/unit_test/TestViewLayoutStrideAssignment.hpp:678: Failure

Death test: { dst = src; }

    Result: died but not with expected error.

  Expected: View assignment must have compatible layouts

Actual msg:

[  DEATH   ] 

[  FAILED  ] TEST_CATEGORYDeathTest.view_layoutstride_right_to_layoutleft_assignment (117 ms)
```

Shouldn't have anything to do with the changes here though...

File: core/src/impl/Kokkos_CPUDiscovery.cpp [hunk 1]
```diff
@@ -55,29 +55,14 @@
 namespace Kokkos {
 namespace Impl {
 
-// The following function (processors_per_node) is copied from here:
-// https://lists.gnu.org/archive/html/autoconf/2002-08/msg00126.html
-// Philip Willoughby
-
 int processors_per_node() {
-#ifdef _WIN32
```
Comment (dalg24) [inline core/src/impl/Kokkos_CPUDiscovery.cpp]: I suppose we should drop https://github.com/kokkos/kokkos/blob/0d1f546f20f44d939e8dac34833266912147c027/core/src/impl/Kokkos_CPUDiscovery.cpp#L44-L46 as well in that case.

**Comment by dalg24** [comment]
@ndellingwood related to #2617

File: core/src/impl/Kokkos_CPUDiscovery.cpp (hunk 1 above)
Comment (crtrott) [reply core/src/impl/Kokkos_CPUDiscovery.cpp]: We are planning to put it back in once we figured out how to do it in windows correctly.

**Comment by dalg24** [comment]
We need this in `develop` as well.

**Pull request merged**